symfony/plugins/orangehrmPimPlugin/Controller/EmployeeReportToController.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Pim\Controller;
+
+use OrangeHRM\Core\Controller\Exception\VueControllerException;
+use OrangeHRM\Core\Exception\DaoException;
+use OrangeHRM\Core\Vue\Component;
+use OrangeHRM\Core\Vue\Prop;
+use OrangeHRM\Framework\Http\Request;
+use OrangeHRM\Pim\Dto\ReportingMethodSearchFilterParams;
+use OrangeHRM\Pim\Service\ReportingMethodConfigurationService;
+
+class EmployeeReportToController extends BaseViewEmployeeController
+{
+
+    /**
+     * @var ReportingMethodConfigurationService|null
+     */
+    protected ?ReportingMethodConfigurationService $reportingMethodService = null;
+
+    /**
+     * @return ReportingMethodConfigurationService
+     */
+    public function getReportingMethodConfigurationService(): ReportingMethodConfigurationService
+    {
+        if (!$this->reportingMethodService instanceof ReportingMethodConfigurationService) {
+            $this->reportingMethodService = new ReportingMethodConfigurationService();
+        }
+        return $this->reportingMethodService;
+    }
+
+    /**
+     * @throws VueControllerException
+     * @throws DaoException
+     */
+    public function preRender(Request $request): void
+    {
+        $empNumber = $request->get('empNumber');
+        if ($empNumber) {
+            $component = new Component('employee-report-to');
+            $reportingMethodParamHolder = new ReportingMethodSearchFilterParams();
+            $reportingMethodsObjectArray = $this->getReportingMethodConfigurationService()->getReportingMethodList($reportingMethodParamHolder);
```

Comment:
https://github.com/orangehrm/orangehrm/blob/5.0/symfony/plugins/orangehrmAdminPlugin/Service/PayGradeService.php#L151-L155

Preferred pattern

Reply:
implemented in preferred manner